### PR TITLE
Remove export of columns function

### PR DESCRIPTION
**Pull request by bkamins**
PR following #1611 so that we can have a look at the consequences.

### CODE REVIEW

File: docs/src/lib/types.md [hunk 1]
```diff
@@ -36,18 +36,21 @@ In the future accessing a single row of a data frame via `getindex` or `view` wi
...
 serves as an iterator over rows of an `AbstractDataFrame`, returning `DataFrameRow` objects.
 
-Similarly, the `eachcol` and `columns` functions return a value of the `DataFrameColumns` type, which
+Similarly, the `eachcol` function returns a value of the `DataFrameColumns` type, which
 serves as an iterator over columns of an `AbstractDataFrame`.
-The difference between the return value of `eachcol` and `columns` is the following:
+The return value can have two concrete types:
 
-* The `eachcol` function returns a value of the `DataFrameColumns{<:AbstractDataFrame, true}` type, which is an
+* If the `eachcol` function is called with `names` argument set to `false` (currently the default,
```
Comment (nalimilan) [inline docs/src/lib/types.md]:
```suggestion
* If the `eachcol` function is called with the `names` argument set to `true` (currently the default,
```

File: src/abstractdataframe/iteration.jl [hunk 2]
```diff
@@ -38,21 +38,24 @@ end
 """
     DataFrameColumns{<:AbstractDataFrame, V} <: AbstractVector{V}
 
-Iterator over columns of an `AbstractDataFrame`.
-If `V` is `Pair{Symbol,AbstractVector}` (which is the case when calling
-[`eachcol`](@link)) then each returned value is a pair consisting of
-column name and column vector. If `V` is `AbstractVector` (a value returned by
-the [`columns`](@link) function) then each returned value is a column vector.
+Iterator over columns of an `AbstractDataFrame` constructed using
+[`eachcol`](@link) function. . If `V` is `Pair{Symbol,AbstractVector}` then each
```
Comment (nalimilan) [inline src/abstractdataframe/iteration.jl]:
```suggestion
[`eachcol(df, true)`](@link). If `V` is a `Pair{Symbol,AbstractVector}` then each
```

File: src/abstractdataframe/iteration.jl [hunk 3]
```diff
@@ -38,21 +38,24 @@ end
...
+returned value is a pair consisting of column name and column vector. If `V` is
+`AbstractVector` (a value returned by the [`columns`](@link) function) then each
```
Comment (nalimilan) [inline src/abstractdataframe/iteration.jl]:
```suggestion
an `AbstractVector` (a value returned by [`eachcol(df, false)`](@link)) then each
```

File: src/abstractdataframe/iteration.jl [hunk 4]
```diff
@@ -67,54 +70,46 @@ julia> df = DataFrame(x=1:4, y=11:14)
...
            maximum(col) - minimum(col)
        end
 2-element Array{Int64,1}:
  3
  3
 ```
 """
-columns(df::T) where T<: AbstractDataFrame =
-    DataFrameColumns{T, AbstractVector}(df)
+eachcol(df::T, names::Bool) where T<: AbstractDataFrame =
```
Comment (nalimilan) [inline src/abstractdataframe/iteration.jl]: I'd use the long function form, and add `@inline` just in case.

Comment (bkamins) [reply src/abstractdataframe/iteration.jl]: OK

File: src/abstractdataframe/iteration.jl [hunk 5]
```diff
@@ -38,21 +38,24 @@ end
 """
     DataFrameColumns{<:AbstractDataFrame, V} <: AbstractVector{V}
 
-Iterator over columns of an `AbstractDataFrame`.
-If `V` is `Pair{Symbol,AbstractVector}` (which is the case when calling
-[`eachcol`](@link)) then each returned value is a pair consisting of
-column name and column vector. If `V` is `AbstractVector` (a value returned by
-the [`columns`](@link) function) then each returned value is a column vector.
+Iterator over columns of an `AbstractDataFrame` constructed using
+[`eachcol(df, true)`](@link) if `V` is a `Pair{Symbol,AbstractVector}`. Then each
```
Comment (nalimilan) [inline src/abstractdataframe/iteration.jl]:
```suggestion
[`eachcol(df, true)`](@ref) if `V` is a `Pair{Symbol,AbstractVector}`. Then each
```
Right?

Comment (bkamins) [reply src/abstractdataframe/iteration.jl]: I will check with documenter - it was your change so I retained it and I never remember for sure what should be used when.

Comment (bkamins) [reply src/abstractdataframe/iteration.jl]: You are right. Fixed.

**Comment by bkamins** [comment]
It should be good for a final check. After the release I will make a second step changing `eachcol` to return only vectors.

**Pull request merged**